From: Nathan Scott <nathans@redhat.com>
Date: Mon, 9 Oct 2017 19:23:27 +1100
Subject: CVE-2017-15047: Add input validity checking to redis cluster config
 slot numbers

Provide a clear error message in the event of a bad slot number
from the redis cluster config file, instead of continuing on and
potentially crashing shortly thereafter when the migrating_slots
arrays are written to.

Resolves https://github.com/antirez/redis/issues/4278
---
 src/cluster.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cluster.c b/src/cluster.c
index a9fedce0c7cf..109b442b58b7 100644
--- a/src/cluster.c
+++ b/src/cluster.c
@@ -243,6 +243,9 @@ int clusterLoadConfig(char *filename) {
                 *p = '\0';
                 direction = p[1]; /* Either '>' or '<' */
                 slot = atoi(argv[j]+1);
+                if (slot < 0 || slot >= CLUSTER_SLOTS) {
+                    serverPanic("Bad slot number in cluster config file");
+                }
                 p += 3;
                 cn = clusterLookupNode(p);
                 if (!cn) {
